djongo/utils.py

```diff
@@ -2,29 +2,43 @@ def encode_key(s: str):
     return s.replace("\\", "\\\\").replace("\$", "\\u0024").replace(".", "\\u002e")
 
 
+encode_keys_cache = {}
 def encode_keys(obj, is_key=False):
+    if is_key:
+        if obj in encode_keys_cache:
+            return encode_keys_cache[obj]
+        if isinstance(obj, int):
+            encode_keys_cache[obj] = str(obj)
+        elif isinstance(obj, str):
+            encode_keys_cache[obj] = encode_key(obj)
+        else:
+            return obj
+        return encode_keys_cache[obj]
     if isinstance(obj, dict):
         return {encode_keys(k, True): encode_keys(v) for k, v in obj.items()}
     if isinstance(obj, list):
         return [encode_keys(v) for v in obj]
-    if isinstance(obj, int) and is_key:
-        return str(obj)
-    if isinstance(obj, str) and is_key:
-        return encode_key(obj)
     return obj
 
 
 def decode_key(s: str):
     return s.replace("\\u002e", ".").replace("\\u0024", "\$").replace("\\\\", "\\")
```

Comment:
Could this be done any faster with regex? i.e. https://stackoverflow.com/a/15175239

Reply:
has similar performance as the current .replace, but seems like perform better a little on average